pkg/ostree/ostree.go

```diff
@@ -228,16 +247,17 @@ func ResolveRef(location, ref string, consumerCerts bool, subs *rhsm.Subscriptio
 // resolved or checked against the repository.
 //
 // If the ref is malformed, the function returns with a RefError.
-func Resolve(source SourceSpec) (CommitSpec, error) {
+func Resolve(source SourceSpec, opt ...ResolveOptionFn) (CommitSpec, error) {
+	options := ResolveOptions{}
+	for _, o := range opt {
+		o(&options)
+	}
```

Comment:
This is a very odd pattern for something that could just be an extension of the `SourceSpec` struct.

Reply:
Yeah well I thought that `SourceSpec` is a type I should not touch. Now that I think about it, I can add fields directly in there that is more simple.